src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import org.apache.commons.lang3.Validate;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.Queries;
+import org.spongepowered.api.data.type.DyeColor;
+
+import java.util.Objects;
+
+public final class Color implements DataSerializable {
+
+    private static final int MASK = 0xff;
+
+    public static final Color BLACK = ofHex(0x000000);
+
+    public static final Color GRAY = ofHex(0x808080);
+
+    public static final Color WHITE = ofHex(0xFFFFFF);
+
+    public static final Color BLUE = ofHex(0x0000FF);
+
+    public static final Color GREEN = ofHex(0x008000);
+
+    public static final Color LIME = ofHex(0x00FF00);
+
+    public static final Color RED = ofHex(0xFF0000);
+
+    public static final Color YELLOW = ofHex(0xFFFF00);
+
+    public static final Color MAGENTA = ofHex(0xFF00FF);
+
+    public static final Color PURPLE = ofHex(0xAA00FF);
+
+    public static final Color DARK_CYAN = ofHex(0x008B8B);
+
+    public static final Color DARK_GREEN = ofHex(0x006400);
+
+    public static final Color DARK_MAGENTA = ofHex(0x8B008B);
+
+    public static final Color NAVY = ofHex(0x000080);
+
+    public static final Color UNKNOWN = ofHex(0xFF00AA);
+
+    public static Color ofHex(int hex) {
```

Comment:
Isn't the format you're accepting here pretty much the standard everywhere, e.g. also used in CSS etc.?